plugins/obs-outputs/flv-mux.c

```diff
@@ -626,7 +626,7 @@ void flv_packet_metadata(enum video_id_t codec_id, uint8_t **output, size_t *siz
 	 * The default trackId is 0.
 	 */
 	if (is_multitrack) {
-		s_w8(&s, MULTITRACKTYPE_ONE_TRACK | PACKETTYPE_METADATA);
+		s_w8(&s, (uint8_t)MULTITRACKTYPE_ONE_TRACK | (uint8_t)PACKETTYPE_METADATA);
```

Comment:
I'm a bit unsure whether it would be more correct to do `(uint8_t)(MULTITRACKTYPE_ONE_TRACK | PACKETTYPE_METADATA)` to ensure the enum values are logically combined with their original type (as intended) and it's the result of the logical `OR` that is then cast to an unsigned 8-bit integer.

The cast itself is correct and consistent with the call on line 632.

Reply:
Ah, that's a fair point. @emaste ?

Reply:
(I believe) oring them before the cast will still fail because they are indeed different types which is what Clang complains about.

Reply:
cc @DimitryAndric

Reply:
Right, my bad - should've looked at the original Clang error in more detail. If I read this right this is partly to meet C++20 deprecation (and C++26 removal) of that "feature".

So the original change is indeed the correct one, even though it works under the (properly correct for the time being) assumption that both enum values cast to `uint8_t` and logically combined will yield the same meaning.

The "proper" fix is to make use of the same `enum` for such operations but that would require a more extensive change.

Reply:
Yeah, I implemented a minimal fix for the FreeBSD port, since we don't want to differ from upstream too much (if possible).